enumerable.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    if block_given?
+      (0..length - 1).my_each do |i|
+        yield(self[i])
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
```

Comment:
- Why are you calling my_each again inside it? :thinking: 

- `[1,2,3].my_each { |x| p x }` should print number 1, 2 and 3. Right now it gives an error: "undefined local variable or method `length' for 0..4:Range". Try to use another way to loop instead of a range.

- Just like the original each method. Your "my_each should return an enumerator if no block was passed.
`irb(main):002:0> [1,2,3,4,5].each`
`=> #<Enumerator: [1, 2, 3, 4, 5]:each>`